java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueVector.java

```diff
@@ -166,13 +168,23 @@ public void setInitialCapacity(int numRecords) {
    *                This helps in tightly controlling the memory we provision
    *                for inner data vector.
    */
+  @Override
   public void setInitialCapacity(int numRecords, double density) {
+    if ((numRecords * density) >= 2_000_000_000) {
```

Comment:
why are we using a different constant here than MAX_ALLOCATION_SIZE?

Reply:
Done to safeguard against truncation. We can use Integer.MAX_VALUE instead.